tests/test_gorest.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from typing import Any
+import json
+from src.gorest import fetch,user_details,user_list
+from tests.mocks.function_gorest import mock_user_details_success,mock_user_list_success
+
+def test_fetch():
+    """ Test case for fetch method to check the response of method call """    
+    response = fetch('https://gorest.co.in/public/v2/users',"https://gorest.co.in/public/v2/users/")
+    assert type(response)==list
+    assert response[0]['id']==3786877
+
+def test_user_list():
+    """ Test case for user_list method to check the response of api call"""    
+    response = user_list('https://gorest.co.in/public/v2/users')
+    assert type(response)==list
+    assert response[0]==3786877
+
+def test_user_details():
+    """ Test case for user_details method to check the response of get call """    
+    response = user_details('https://gorest.co.in/public/v2/users/',[3786877])
+    assert type(response)==list
+    assert response[0]['id']== 3786877
```

Comment:
Alright, these test cases are dependent on internet to work with actual apis. Here it is making actual requests which we dont need to do so. All we need to do is to test whether your methods are working as per expected or not. Mocks these and writing testcases would be fine.
NOTE: This is the riight way, yet these are dependent. Just explained this way of api test cases for learning purpose